test/behaviour/connection/ConnectionSteps.java

```diff
@@ -74,14 +77,23 @@ public synchronized void before() {
         String address = GraknSingleton.getGraknRunner().address();
         assertNotNull(address);
         client = new GraknClient(address);
+        client.databases().all().forEach(database -> client.databases().delete(database));
         System.out.println("ConnectionSteps.before");
     }
 
     @After
     public synchronized void after() {
         sessions.parallelStream().forEach(Session::close);
         sessions.clear();
+
+        Stream<CompletableFuture<Void>> closures = sessionsParallel
+                .stream().map(futureSession -> futureSession.thenApplyAsync(session -> {
+                    session.close();
+                    return null;
+                }));
+        CompletableFuture.allOf(closures.toArray(CompletableFuture[]::new)).join();
```

Comment:
We were closing `sessions`, but not `sessionsParallel`. Tests that used `sessionsParallel` were simply leaving behind lingering sessions.